src/connection.ts

```diff
@@ -269,13 +276,18 @@ export class Connection extends JSForceConnection {
       this.query<T>(soql, _options)
         .on('record', rec => records.push(rec))
         .on('error', err => reject(err))
-        .on('end', () =>
+        .on('end', () => {
+          if (records.length >= limit) {
```

Comment:
I thought there was a `totalSize` on the query that would return the total number of records even if it wasn't retrieved. For example. I run a query and it matches 20k records on Salesforce. The first response will return a `totalSize` of 20k, with 2k records, and query locators to get the next 18k records in 9 more responses. That would mean if the limit was 10k, `totalSize` would be 20k, `records.length` would be 10k, and `limit` would be 10k. So the `totalSize > records.length` would tell is if missing data is there.

However, that isn't what I saw when I ran a few tests manually... so I must have something wrong or jsforce is masking something. In any case, this is close enough to equivalent that it is probably fine.

Reply:
I think that the `totalSize` field we're used to seeing in toolbelt is coming from below. `totalSize: records.length`